prow/plugins/jira-linker/jira-linker_test.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jira_linker
+
+import (
+	"testing"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/github/fakegithub"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+func TestJiraLink(t *testing.T) {
+	const jiraBaseUrl = "http://example.com/jira"
+
+	repoLabelCommentPrefix := "org/repo#5:"
+
+	for _, tc := range []struct {
+		name                 string
+		title                string
+		hasLabels            []string
+		labelsAddedOverall   []string
+		labelsRemovedOverall []string
+		shouldComment        bool
+		comment              string
+	}{
+		{
+			name:               "non-jira PR",
+			title:              "add sauce to spaghetti",
+			hasLabels:          []string{},
+			labelsAddedOverall: []string{noJiraLabel},
+			shouldComment:      false,
+		},
+		{
+			name:               "basic jira PR",
+			title:              "TEST-43 add sauce to spaghetti",
+			hasLabels:          []string{},
+			labelsAddedOverall: []string{jiraPrefix + "TEST"},
+			shouldComment:      true,
+			comment:            commentForTicket(jiraLink(jiraBaseUrl, "TEST-43")),
+		},
+		{
+			name:                 "starts with wrong labels",
+			title:                "TEST-32 fix every bug",
+			hasLabels:            []string{noJiraLabel},
+			labelsAddedOverall:   []string{noJiraLabel, jiraPrefix + "TEST"},
+			labelsRemovedOverall: []string{noJiraLabel},
+			shouldComment:        true,
+			comment:              commentForTicket(jiraLink(jiraBaseUrl, "TEST-32")),
+		},
+		{
+			name:               "starts with right labels",
+			title:              "TEST-32 fix every bug",
+			hasLabels:          []string{jiraPrefix + "TEST"},
+			labelsAddedOverall: []string{jiraPrefix + "TEST"},
+			shouldComment:      false,
+		},
```

Comment:
Perhaps a test for the case Pete mentioned, where there are two ticket strings in the title?